pennylane/transforms/combine_global_phases.py

```diff
@@ -0,0 +1,73 @@
+"""
+Provides a transform to combine all ``qml.GlobalPhase`` gates in a circuit into a single one applied at the end.
+"""
+
+import pennylane as qml
+from pennylane.tape import QuantumScript, QuantumScriptBatch
+from pennylane.transforms import transform
+from pennylane.typing import PostprocessingFn
+
+
+@transform
+def combine_global_phases(tape: QuantumScript) -> tuple[QuantumScriptBatch, PostprocessingFn]:
+    """Combine all ``qml.GlobalPhase`` gates into a single ``qml.GlobalPhase`` operation.
+
+    Circuits passed through this transform will be updated to include the same operations and
```

Comment:
Technically the transform returns a new circuit. It does not update the original circuit. You might want to rephrase this to avoid confusion.

Reply:
Yes, thank you for the comment.
I just fixed this in commit https://github.com/PennyLaneAI/pennylane/pull/6686/commits/fde6ee8db09b82ee80baa33bc4f090a814c20bc4.